File: Java/AD5241.java
// Distributed with a free-will license.
// Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
// AD5241
// This code is designed to work with the AD5241_I2CPOT_10K I2C Mini Module available from ControlEverything.com.
// https://www.controleverything.com/content/Potentiometers?sku=AD5241_I2CPOT_10K#tabs-0-product_tabset-2

import com.pi4j.io.i2c.I2CBus;
import com.pi4j.io.i2c.I2CDevice;
import com.pi4j.io.i2c.I2CFactory;
import java.io.IOException;

public class AD5241
{
	public static void main(String args[]) throws Exception
	{
		// Create I2C bus
		I2CBus Bus = I2CFactory.getInstance(I2CBus.BUS_1);
		// Get I2C device, AD5241 I2C address is 0x2C(44)
		I2CDevice device = Bus.getDevice(0x2C);

		// Send command byte
		// Input resistance value, 0x80(128)
		device.write(0x00, (byte)0x80);
		Thread.sleep(500);

		// Convert the data
		byte data = (byte)device.read();
		double resistance = ((data & 0xFF) / 256.0) * 10.0;

		// Output data to screen
		System.out.printf("Resistance %.2f K", resistance);
	}
}

File: Python/AD5241.py
# Distributed with a free-will license.
# Use it any way you want, profit or free, provided it fits in the licenses of its associated works.
# AD5241
# This code is designed to work with the AD5241_I2CPOT_10K I2C Mini Module available from ControlEverything.com.
# https://www.controleverything.com/content/Potentiometers?sku=AD5241_I2CPOT_10K#tabs-0-product_tabset-2

import smbus
import time

# Get I2C bus
bus = smbus.SMBus(1)

# AD5241 address, 0x2C(44)
# Send command byte, 0x00(00)
#		0x80(128)	Input resistance value
bus.write_i2c_block_data(0x2C, 0x00, [0x80])

time.sleep(0.5)

# AD5241 address, 0x2C(44)
# Read data back from 0x00(00), 1 byte
data = bus.read_byte_data(0x2C, 0x00)

# Convert the data
resistance = (data / 256.0 ) * 10.0

# Output data to screen
print "Resistance : %.2f K" %resistance
